aws/resource_aws_s3_bucket_object.go

```diff
@@ -366,7 +366,7 @@ func resourceAwsS3BucketObjectRead(d *schema.ResourceData, meta interface{}) err
 	// The "STANDARD" (which is also the default) storage
 	// class when set would not be included in the results.
 	d.Set("storage_class", s3.StorageClassStandard)
-	if resp.StorageClass != nil {
+	if resp.StorageClass != nil { // nosem: helper-schema-ResourceData-Set-extraneous-nil-check
```

Comment:
And this is the same logic as its datasource counterpart, so checks could be reversed.